c_src/mzmetrics_nif.c

```diff
@@ -203,7 +203,7 @@ static ERL_NIF_TERM metrics_update_resource_counter_nif(ErlNifEnv* env, int argc
 	    return enif_make_badarg(env);
     if (key >= resource->counter->num_cnt)
 	    return enif_make_badarg(env);
-    if (!enif_get_int(env, argv[2], &val))
+    if (!enif_get_int64(env, argv[2], &val))
```

Comment:
So based on the crash the thesis was that the overflow caused `enif_get_int/3` to return false and make thus the function creates a `badarg` exception. That makes sense to me.